tools/check-cppcheck.sh

```diff
@@ -15,7 +15,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-CPPCHECK_JOBS=${CPPCHECK_JOBS:=$(nproc)}
+if [[ "$OSTYPE" == "linux"* ]]; then
+    CPPCHECK_JOBS=${CPPCHECK_JOBS:=$(nproc)}
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+    CPPCHECK_JOBS=${CPPCHECK_JOBS:=$(sysctl -n hw.ncpu)}
```

Comment:
Thanks for getting this work on OSX! This script becomes just a little bit incomplete, however. Please, add a final `else` branch that sets `CPPCHECK_JOBS` to something basic default, just to make sure that the script doesn't break. (E.g., if someone builds on freebsd, or whatever.) I'd go for a default of 1.